pkg/kustomize/kustomize.go

```diff
@@ -41,6 +41,13 @@ type Kustomizer interface {
 	// - applying the customization fails
 	// As of the current implementation, it is not safe to call this function concurrently
 	ApplyLabels(resourceUri *url.URL, labels map[string]string) ([]byte, error)
+
+	// Applies the provided labels to the resource definition
+	// Returns the customized resource contents
+	// Returns an error if
+	// - applying the customization fails
+	// As of the current implementation, it is not safe to call this function concurrently
+	ApplyLabelsOnContent(resourceDefinition string, labels map[string]string) ([]byte, error)
```

Comment:
why do we need this and ApplyLabels? Perhaps consolidate both methods with this signature as `ApplyLabels`

Reply:
done!